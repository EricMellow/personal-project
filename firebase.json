{
  "hosting": {
    "public": "/",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**",
      "keys.js"
    ]
  }
}
