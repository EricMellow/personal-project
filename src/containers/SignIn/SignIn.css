.sign-in {
  align-items: center;
  background-repeat: no-repeat;
  background-size: cover;
  background: url(../../assets/bricks1.jpg); 
  display: flex;
  flex-direction: column;
  height: 85vh;
  justify-content: center;
}

.sign-in-form {
  align-items: center;
  background-color: rgba(28, 141, 196, .9);
  display: flex;
  flex-direction: column;
  height: 55%;
  justify-content: center;
  width: 50%;
}

.input {
  background: none;
  border-radius: 5px;
  border: 3px solid white;
  color: white;
  font-size: 2rem;
  height: 3rem;
  margin: 20px;
  padding-left: 10px;
  width: 75%;
}

.sign-in-btn {
  background: none;
  border-radius: 5px;
  border: 2px solid white;
  color: white;
  font-size: 2rem;
  height: 3rem;
  margin: 20px;
  width: 75%;
}

.sign-in-btn:disabled {
  background: rgba(255, 255, 255, 0.15);
  border: 2px solid rgb(67, 119, 144);
  color: rgb(67, 119, 144);
}

*, *:before, *:after {
  box-sizing: inherit;
}

form p {
  color: white;
  font-size: 1.2rem;
  margin: 0;
}