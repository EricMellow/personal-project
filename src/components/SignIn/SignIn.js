import React, { Component } from "react";
import './SignIn.css';
import { auth } from '../../firebase';

export class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      error: null
    };
  }

  handleInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  resetState = () => {
    this.setState({
      email: '',
      password: '',
      error: null
    });
  }

  storeData = async (event) => {
    event.preventDefault();
    try {
      await auth.doSignInWithEmailAndPassword(this.state.email, this.state.password);
      this.resetState();
      this.props.history.push('/distance');
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const {
      email,
      password,
      error
    } = this.state;

    const invalidUserInfo =
      password === '' ||
      email === '';

    return (
      <form onSubmit={event => this.storeData(event)}>
        <input
          placeholder="Email Address"
          name="email"
          className="input email-input"
          value={email}
          onChange={event => this.handleInput(event)}
          type="text"
        />
        <input
          placeholder="Password"
          name="password"
          className="input password-input"
          value={password}
          onChange={event => this.handleInput(event)}
          type="password"
        />
        <button type="submit" disabled={invalidUserInfo}>
          Sign In
        </button>

        {error && <p>{error}</p>}
      </form>
    );
  }
}

export default SignIn;