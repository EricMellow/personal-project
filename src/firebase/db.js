import { db } from './firebase';

// User API

export const doCreateUser = (id, username, email, zipcode) =>
  db.ref(`users/${id}`).set({
    username,
    email,
    zipcode
  });

export const doCreateActivity = (address, duration, lat, lng, type) => 
  db.ref('actions/').push({
    address,
    type,
    duration,
    lat,
    lng
  })

export const doGetAllActivities = () => 
  db.ref('actions/').on('value', snapshot => snapshot.val())

// export const onceGetUsers = () =>
//   db.ref('users').once('value');