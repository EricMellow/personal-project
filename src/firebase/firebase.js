import { firebaseKey } from '../keys';
import * as firebase from 'firebase';
const key = firebaseKey;
// Initialize Firebase
var config = {
  apiKey: key,
  authDomain: "personal-project-1527122816740.firebaseapp.com",
  databaseURL: "https://personal-project-1527122816740.firebaseio.com",
  projectId: "personal-project-1527122816740",
  storageBucket: "",
  messagingSenderId: "182263229649"
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const db = firebase.database();
const auth = firebase.auth();

export { db, auth };