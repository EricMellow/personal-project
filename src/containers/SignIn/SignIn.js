import React, { Component } from "react";
import './SignIn.css';
import { auth } from '../../firebase';
import { connect } from 'react-redux';
import { authenticateUser } from "../../actions/authenticateUser";
import { addUserId } from "../../actions/userIdActions";
import { addZipcode } from "../../actions/zipcodeActions";
import * as firebase from "../../firebase/firebase";

export class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      error: null
    };
  }

  handleInput = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  resetState = () => {
    this.setState({
      email: '',
      password: '',
      error: null
    });
  }

  storeData = async (event) => {
    event.preventDefault();
    try {
      const authUser = await auth.doSignInWithEmailAndPassword(this.state.email, this.state.password);
      this.props.storeUserId(authUser.user.uid);
      this.resetState();
      this.props.authenticate();
      firebase.db.ref(`users/${authUser.user.uid}`).once('value')
        .then(snapshot => this.props.storeZipcode(snapshot.val().zipcode));
      this.props.history.push('/distance');
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    const {
      email,
      password,
      error
    } = this.state;

    const invalidUserInfo =
      password === '' ||
      email === '';

    return (
      <div className="sign-in">
        <form className="sign-in-form" onSubmit={event => this.storeData(event)}>
          <input
            placeholder="Email Address"
            name="email"
            className="input email-input"
            value={email}
            onChange={event => this.handleInput(event)}
            type="text"
          />
          <input
            placeholder="Password"
            name="password"
            className="input password-input"
            value={password}
            onChange={event => this.handleInput(event)}
            type="password"
          />
          <button 
            type="submit" 
            className="sign-in-btn"
            disabled={invalidUserInfo}
          >
            Sign In
          </button>

          {error && <p>{error}</p>}
        </form>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  authenticate: () => dispatch(authenticateUser()),
  storeUserId: (userId) => dispatch(addUserId(userId)),
  storeZipcode: (zipcode) => dispatch(addZipcode(zipcode))
});

export default connect(null, mapDispatchToProps)(SignIn);