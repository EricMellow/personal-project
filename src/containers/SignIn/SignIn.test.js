import { SignIn, mapDispatchToProps } from "./SignIn";
import { shallow } from "enzyme";
import React from "react";
import { auth } from "../../firebase";


describe('SignIn', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SignIn />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleInput', () => {

    it('should set the correct key/value pair in state with the incomming event data', () => {
      const wrapper = shallow(<SignIn />);
      const mockEvent = {
        target: {
          name: 'password',
          value: 'trustnoone'
        }
      };
      const initialExpected = {
        email: '',
        password: '',
        error: null
      };
      const finalExpected = {
        email: '',
        password: 'trustnoone',
        error: null
      };

      expect(wrapper.state()).toEqual(initialExpected);
      wrapper.instance().handleInput(mockEvent);
      expect(wrapper.state()).toEqual(finalExpected);
    });
  });

  describe('resetState', () => {

    it('should reset the state to the default settings', () => {
      const wrapper = shallow(<SignIn />);
      const initialState = {
        email: 'test@test.com',
        password: 'password',
        error: null
      };
      wrapper.setState(initialState);
      const expected = {
        email: '',
        password: '',
        error: null
      };

      expect(wrapper.state()).toEqual(initialState);
      wrapper.instance().resetState();
      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe('storeData', () => {
    let wrapper;
    let mockHistory;
    let mockEvent;
    let mockAuth;
    let mockStoreId;

    beforeEach(() => {
      mockEvent = { preventDefault: jest.fn() };
      mockHistory = { push: jest.fn() };
      mockAuth = jest.fn();
      mockStoreId = jest.fn();
      wrapper = shallow(<SignIn 
        history={mockHistory}
        storeUserId={mockStoreId}
        authenticate={mockAuth}
      />);
      auth.doSignInWithEmailAndPassword = jest.fn().mockImplementation(() => Promise.resolve({user: {uid: 12345}}));
    });

    it('should call doSignInWithEmailAndPassword with the correct arguments', () => {
      wrapper.setState({
        email: 'test@test.com',
        password: 'trustnoone'
      });

      wrapper.instance().storeData(mockEvent);
      expect(auth.doSignInWithEmailAndPassword).toHaveBeenCalledWith('test@test.com', 'trustnoone');
    });

    it('should reset the state to the default parameters', async () => {
      const initialState = {
        email: 'test@test.com',
        password: 'password',
        error: null
      };
      wrapper.setState(initialState);
      const expected = {
        email: '',
        password: '',
        error: null
      };

      expect(wrapper.state()).toEqual(initialState);
      await wrapper.instance().storeData(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });

    // it('should set the error key in state with a value of the error message if there is an error', () => {
    //   auth.doCreateUserWithEmailAndPassword = jest.fn().mockImplementation(() => Promise.reject(new Error({message: 'oops!'})));
    //   const expected = 'oops!';

    //   wrapper.instance().storeData(mockEvent);
    //   expect(wrapper.state('error')).toEqual(expected);

    // });

  });

  describe('return', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<SignIn />);
      wrapper.instance().storeData = jest.fn();
      wrapper.instance().handleInput = jest.fn();
    });

    it('should call storeData on submit', () => {
      wrapper.find('form').simulate('submit');
      expect(wrapper.instance().storeData).toHaveBeenCalled();
    });

    it('should call handleInput on change of the email input', () => {
      wrapper.find('input.email-input').simulate('change');
      expect(wrapper.instance().handleInput).toHaveBeenCalled();
    });

    it('should call handleInput on change of the password input', () => {
      wrapper.find('input.password-input').simulate('change');
      expect(wrapper.instance().handleInput).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    
    it('should call dispatch on authenticateUser with the correct arguments', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'AUTHENTICATED_USER'
      };

      mappedProps.authenticate();
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('should call dispatch on AddUserId with the correct arguments', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'ADD_USER_ID',
        userId: 'abc123'
      };

      mappedProps.storeUserId('abc123');
      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    });

    it('should call dispatch on addZipcode with the correct arguments', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'ADD_ZIPCODE',
        zipcode: 90210
      };

      mappedProps.storeZipcode(90210);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction)
    });
  });

});