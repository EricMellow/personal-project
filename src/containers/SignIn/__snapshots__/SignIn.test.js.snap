// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="input email-input"
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          type="text"
          value=""
/>,
        <input
          className="input password-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
/>,
        <button
          className="sign-in-btn"
          disabled={true}
          type="submit"
>
          Sign In
</button>,
        null,
      ],
      "className": "sign-in",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input email-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email Address",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input password-input",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
          "className": "sign-in-btn",
          "disabled": true,
          "type": "submit",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "button",
      },
      null,
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input email-input"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="text"
            value=""
/>,
          <input
            className="input password-input"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <button
            className="sign-in-btn"
            disabled={true}
            type="submit"
>
            Sign In
</button>,
          null,
        ],
        "className": "sign-in",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input email-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input password-input",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "className": "sign-in-btn",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "button",
        },
        null,
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
